pkg/generate/app/cmd/newapp.go

```diff
@@ -418,10 +418,21 @@ type AppResult struct {
 
 	BuildNames []string
 	HasSource  bool
+	Namespace  string
 }
 
-// Run executes the provided config.
-func (c *AppConfig) Run(out io.Writer) (*AppResult, error) {
+// RunAll executes the provided config to generate all objects.
+func (c *AppConfig) RunAll(out io.Writer) (*AppResult, error) {
+	return c.run(out, app.Acceptors{app.NewAcceptUnique(c.typer), app.AcceptNew})
+}
+
+// RunBuilds executes the provided config to generate just builds.
+func (c *AppConfig) RunBuilds(out io.Writer) (*AppResult, error) {
+	return c.run(out, app.Acceptors{app.NewAcceptBuildConfigs(c.typer), app.NewAcceptUnique(c.typer), app.AcceptNew})
+}
+
+// run executes the provided config applying provided acceptors.
+func (c *AppConfig) run(out io.Writer, acceptors app.Acceptors) (*AppResult, error) {
```

Comment:
Line 444 - if you asked for NewBuild and you don't have source you should return an error.